Fix member variable ordering in mesh.h

A recent change to `mesh.h` doesn't obey the `-Werror=reorder` compile flag:

```
/ramdisk/civet/build_0/cardinal/contrib/openmc/include/openmc/mesh.h: In constructor 'openmc::StructuredMesh::MeshDistance::MeshDistance(int, bool, double)':
/ramdisk/civet/build_0/cardinal/contrib/openmc/include/openmc/mesh.h:149:10: error: 'openmc::StructuredMesh::MeshDistance::max_surface' will be initialized after [-Werror=reorder]
  149 |     bool max_surface {true};
      |          ^~~~~~~~~~~
/ramdisk/civet/build_0/cardinal/contrib/openmc/include/openmc/mesh.h:148:12: error:   'double openmc::StructuredMesh::MeshDistance::distance' [-Werror=reorder]
  148 |     double distance {INFTY};
      |            ^~~~~~~~
/ramdisk/civet/build_0/cardinal/contrib/openmc/include/openmc/mesh.h:144:5: error:   when initialized here [-Werror=reorder]
  144 |     MeshDistance(int _index, bool _max_surface, double _distance)
```

MOOSE uses this flag, which is preventing Cardinal from building with the latest OpenMC.

--- a/include/openmc/mesh.h
+++ b/include/openmc/mesh.h
@@ -145,8 +145,8 @@ public:
       : next_index {_index}, max_surface {_max_surface}, distance {_distance}
     {}
     int next_index {-1};
-    double distance {INFTY};
     bool max_surface {true};
+    double distance {INFTY};
     bool operator<(const MeshDistance& o) const
     {
       return distance < o.distance;

